### PR TITLE
rename sub/slice -> view

### LINKED ISSUE #16846: Change sub behaviour to match getindex, possibly rename.
Following #13612, we should change `sub` behaviour to match which, as I understand it, basically means renaming `slice` to `sub` (and deprecating `slice`).

There was some discussion in #13157 of changing the name to `view` (which would conflict with ArrayViews), but if so we should make that decision soon.

cc: @andreasnoack @timholy @mbauman

### PR DESCRIPTION
**Pull request by simonbyrne**
fixes #16846

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -5402,14 +5402,14 @@ is raised as an exception in the woken tasks.
 notify
 
 """
-    sub(A, inds...)
+    view(A, inds...)
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: rst docs too

Comment (simonbyrne) [inline base/docs/helpdb/Base.jl]: done.

**Comment by simonbyrne** [comment]
Okay, looks like this is now working.

**Comment by timholy** [comment]
:+1: Of course, note that we have multiple view types now: `ReshapedArray` and the unexported `PermutedDimsArray` being two new ones for julia-0.5. Still, I think `SubArray` is the canonical view, so this is fine.

**Comment by StefanKarpinski** [comment]
Should `SubArray` be renamed to `ArrayView` while we're at it or would that be too disruptive?

**Comment by vtjnash** [comment]
Stefan, I think we should worry about that in v0.6 as we do a more bottom-up refactoring.

**Comment by StefanKarpinski** [comment]
Sounds good to me.

**Comment by simonbyrne** [comment]
Any idea why appveyor is failing?

**Comment by tkelman** [comment]
https://github.com/JuliaLang/julia/issues/16556

**Pull request merged**